v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/transforms/SourceWriterFn.java

```diff
@@ -156,7 +153,7 @@ public void processElement(ProcessContext c) {
       // no shard found, move to permanent error
       outputWithTag(
           c, Constants.PERMANENT_ERROR_TAG, Constants.SHARD_NOT_PRESENT_ERROR_MESSAGE, spannerRec);
-    } else if (shardId.equals(skipDirName)) {
+    } else if (shardId != null && shardId.equals(skipDirName)) {
```

Comment:
There is already a check for shardId== null above, why do we need additional check here?